FIX: correct bug in _concatenate_info function

hstojic opened the pull request:
This is related to issue #2844 that I raised, more details is provided there. 

In summary, using `concatenate_runs = True` results in an error as regressor names are missing for session dummy variables, this one-line corrects it by adding a session regressor name to the regressor_names variable..

If my interpretation is correct, its an easy fix with a single line I think:
`infoout.regressor_names.extend(['run' + str(i + 1)])`

hstojic committed:
correct bug in _concatenate_info function

Using concatenate_runs = True results in an error as regressor names are missing for session dummy variables, this one-line corrects it.

effigies reviewed (dismissed):
This looks sensible. I think it could be written slightly more cleanly. And can you verify that this produces the results you expect?

And do you want to add yourself to the [`.zenodo.json`](https://github.com/nipy/nipype/blob/master/.zenodo.json) file?

hstojic commented:
Yes, this produces the results I expect, I already use it in my local code.

I will add myself to the zenodo file, thanks!

effigies commented:
Just noticed the failing tests. Looks like `infoout.regressor_names` doesn't exist to extend. If you look at lines 563-564, you need to create the list first. I'm a little curious why it works for you, then...

I'll try to have a deeper look tomorrow.

effigies commented:
Also, can you go ahead and merge/cherry-pick 4158583 into this PR instead of having a separate one?

hstojic commented:
Ok, silly mistake, `regressors` is an optional argument and so is `regressor_names` then. In my case I had nuisance regressors and their names so it never complained. Tests for sure have cases without any nuisance covariate. should be Ok by just adding creating the list few lines above.

hstojic committed:
creating infout.regressor_names list

corrected and error in _concatenate_info - when regressor and regressor_names are not defined, the fix fails. Now the regressor_names are created if they were not defined by the user.

satra commented:
@hstojic  and @effigies - since this creates a patch without a broken test, it would be great to have a broken test as well. i just asked @hstojic to add that in the original issue.

hstojic committed:
Added Hrvoje Stojic details to contributors list

hstojic commented:
@satra and @effigies I have been a bit tight on time these few days to add a test and resolve the issue before the new release, it will have to wait for the new one, I'll try to do it soonish

effigies commented:
No worries. When you get some time.

effigies commented:
Hi @hstojic. Just a bump to let you know we'll be aiming to have everything in for the 1.1.9 release by next Friday, Feb 22. Let us know if you need any help on this.

effigies committed:
Merge branch 'master' into patch-1
